src/update_agent/actor.rs

```diff
@@ -169,30 +169,49 @@ impl UpdateAgent {
         }
     }
 
+    /// Attempt to register as the update driver for rpm-ostree.
+    fn register_as_driver(actor: &UpdateAgent) -> ResponseActFuture<UpdateAgent, Result<(), ()>> {
+        let msg = rpm_ostree::RegisterAsDriver {};
+        let result = actor
+            .rpm_ostree_actor
+            .send(msg)
+            .unwrap_or_else(|e| Err(e.into()))
+            .map_err(|e| log::error!("failed to register as driver: {}", e))
+            .into_actor(actor);
+
+        Box::pin(result)
+    }
+
     /// Initialize the update agent.
     fn tick_initialize(&mut self) -> ResponseActFuture<Self, Result<(), ()>> {
         trace!("update agent in start state");
 
-        let initialization = self.local_deployments().map(|res, actor, _ctx| {
-            let status;
-            if actor.enabled {
-                status = "initialization complete, auto-updates logic enabled";
-                log::info!("{}", status);
-                actor.state.initialized();
-                actor.strategy.record_details();
+        let initialization = self
+            .local_deployments()
+            .then(|res, actor, _ctx| {
                 // Report non-future update target deployments.
                 if let Ok(depls) = res {
                     Self::log_excluded_depls(&depls, actor);
                 }
-            } else {
-                status = "initialization complete, auto-updates logic disabled by configuration";
-                log::warn!("{}", status);
-                actor.state.end();
-            };
-            update_unit_status(status);
-            notify_ready();
-            Ok(())
-        });
+                Self::register_as_driver(actor)
+            })
+            .map(|_r, actor, _ctx| {
```

Comment:
I'm not sure if this is the right thing to do. We don't actually need to wait for the results of `local_deployments()` before we `register_as_driver()`, so chaining like this seems unnecessary. But OTOH, I'm not sure how to combine `local_deployments()` and `register_as_driver()` into one elegantly. 
@lucab

Reply:
Hmm or does all of this not really matter since we're really just chaining all the actions within a tick into one big future..